android/app/src/main/java/com/reactnativenavigation/controllers/NavigationActivity.java

```diff
@@ -151,6 +153,7 @@ protected void onDestroy() {
         destroyLayouts();
         destroyJsIfNeeded();
         NavigationApplication.instance.getActivityCallbacks().onActivityDestroyed(this);
+        numberOfActivities -= 1;
```

Comment:
I noticed you're incrementing activity count in onCreate and decrementing in onDestroy. This doesn't seem like what you want as the activity can be recreated after onStop

Reply:
You’re right, I was looking at the android [Activity Lifecycle](https://developer.android.com/guide/components/activities/activity-lifecycle.html) and I am afraid I can’t find any way to reliably use this counter.